### PR TITLE
[IMP] discuss: start-a-meeting improvements

### PR DESCRIPTION
**Pull request by oomsveta**
- Improve responsivity of WelcomeView and MediaPreview
- Reuse MediaTrackConstraints already defined in mail.rtc
- Properly display guest's name when starting a live conference

### CODE REVIEW

File: addons/mail/models/mail_channel.py [hunk 1]
```diff
@@ -477,8 +477,8 @@ def is_current_user(r):
                     },
                 },
             ])
-        notification = _('%s started a live conference', partner.name)
-        self.message_post(body=notification, message_type="notification")
+        notification = _("%s started a live conference", partner.name if partner else guest.name)
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:480]: `partner.name or guest.name` should do the job.

Note that just this change would be more easily merged as a separate PR.

File: addons/mail/static/src/components/welcome_view/welcome_view.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
+.o_WelcomeView {
+    min-height: 100%;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/welcome_view/welcome_view.scss:2]: What's the change between using height and min-height in this case? If it's because parent is flex, this should have flex-grow-1 instead.

File: addons/mail/static/src/components/welcome_view/welcome_view.xml [hunk 3]
```diff
@@ -2,12 +2,19 @@
...
     <t t-name="mail.WelcomeView" owl="1">
-        <div class="o_WelcomeView h-100 d-flex flex-column justify-content-center align-items-center bg-light">
-            <h1 class="font-weight-light">You've been invited to a meeting!</h1>
-            <h2 class="m-5" t-esc="messaging.companyName"/>
-            <div class="d-flex justify-content-center">
-                <MediaPreview class="mr-5" localId="welcomeView.mediaPreview.localId"/>
-                <div class="d-flex flex-column justify-content-center">
+        <div class="o_WelcomeView d-flex flex-column align-items-center bg-light" t-att-class="{ 'justify-content-center': !messaging.device.isMobile }">
+            <t t-if="!messaging.device.isMobile">
+                <h1 class="font-weight-light">You've been invited to a meeting!</h1>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/welcome_view/welcome_view.xml:7]: When the window is really small (or the text gets longer with translation) this (and other elements) should take some margin to avoid being too close to border.

File: addons/mail/static/src/components/welcome_view/welcome_view.xml [hunk 4]
```diff
@@ -2,12 +2,19 @@
...
+                <h2 class="m-5" t-esc="messaging.companyName"/>
+            </t>
+            <div class="d-flex justify-content-around flex-wrap">
+                <MediaPreview localId="welcomeView.mediaPreview.localId"/>
+                <div t-if="!messaging.device.isMobile" class="o_WelcomeView_separator m-5"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/welcome_view/welcome_view.xml:12]: Move as t-att-class on above or below element.

File: addons/mail/static/src/components/welcome_view/welcome_view.xml [hunk 5]
```diff
@@ -2,12 +2,19 @@
...
+                <div class="d-flex flex-column justify-content-center mt-4">
+                    <t t-if="messaging.device.isMobile">
+                        <h1 class="font-weight-light">You've been invited to a meeting!</h1>
+                    </t>
+                    <div t-if="messaging.device.isMobile" class="o_WelcomeView_separator mb-5"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/welcome_view/welcome_view.xml:17]: Move as t-att-class on above or below element.

File: addons/mail/static/src/components/welcome_view/welcome_view.xml [hunk 6]
```diff
@@ -2,12 +2,19 @@
...
+                <h1 class="font-weight-light">You've been invited to a meeting!</h1>
+                <h2 class="m-5" t-esc="messaging.companyName"/>
+            </t>
+            <div class="d-flex justify-content-around flex-wrap">
+                <MediaPreview localId="welcomeView.mediaPreview.localId"/>
+                <div t-if="!messaging.device.isMobile" class="o_WelcomeView_separator m-5"/>
+                <div class="d-flex flex-column justify-content-center mt-4">
+                    <t t-if="messaging.device.isMobile">
+                        <h1 class="font-weight-light">You've been invited to a meeting!</h1>
+                    </t>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/welcome_view/welcome_view.xml:16]: Do we really want the title below in mobile? To validate with CPE

File: addons/mail/static/src/models/media_preview/media_preview.js [hunk 7]
```diff
@@ -148,6 +148,29 @@ function factory(dependencies) {
             return this.videoStream !== null;
         }
 
+        /**
+         * @private
+         * @returns {number}
+         */
+        _computeVideoElementHeight() {
+            return this.videoElementWidth * .75;
```
Comment (seb-odoo) [inline addons/mail/static/src/models/media_preview/media_preview.js:156]: Comment what ratio is this supposed to be, and why pick this one in particular?

Comment (seb-odoo) [reply addons/mail/static/src/models/media_preview/media_preview.js:156]: Probably safer to round it to the nearest integer to avoid strange issues.

File: addons/mail/static/src/models/media_preview/media_preview.js [hunk 8]
```diff
@@ -69,7 +69,7 @@ function factory(dependencies) {
             if (!this.doesBrowserSupportMediaDevices) {
                 return;
             }
-            const audioStream = await navigator.mediaDevices.getUserMedia({ audio: true });
+            const audioStream = await navigator.mediaDevices.getUserMedia({ audio: this.messaging.userSetting.getAudioConstraints() });
```
Comment (seb-odoo) [inline addons/mail/static/src/models/media_preview/media_preview.js:72]: To double check with @ThanhDodeurOdoo why we might want to use those methods.

File: addons/mail/static/src/models/media_preview/media_preview.js [hunk 9]
```diff
@@ -148,6 +148,29 @@ function factory(dependencies) {
...
+         */
+        _computeVideoElementWidth() {
+            const device = this.messaging && this.messaging.device;
+            if (!device) {
+                return 0;
+            }
+            if (device.isMobile) {
+                return device.globalWindowInnerWidth;
+            }
+            return device.globalWindowInnerWidth / 3;
```
Comment (seb-odoo) [inline addons/mail/static/src/models/media_preview/media_preview.js:171]: This might get a bit too small in "almost mobile" size (and probably way too big for people with massive widescreen), so this should have more thresholds for a better result, and a hard limit both up and down.

File: addons/mail/static/src/models/media_preview/media_preview.js [hunk 10]
```diff
@@ -173,6 +196,12 @@ function factory(dependencies) {
         isVideoEnabled: attr({
             compute: '_computeIsVideoEnabled',
         }),
+        videoElementHeight: attr({
+            compute: '_computeVideoElementHeight',
+        }),
+        videoElementWidth: attr({
+            compute: '_computeVideoElementWidth'
+        }),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/media_preview/media_preview.js:204]: We've been a bit lazy recently but ideally we would want all fields to have a docstring.

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Please see https://www.odoo.com/documentation/master/developer/misc/other/guidelines.html#commit-message-structure for commit guideline.
Also this should be [FIX] not [IMP].

**Comment by oomsveta** [comment]
outdated

see odoo/odoo#76388 and odoo/odoo#76226
